### PR TITLE
MULE-16437: Improve ExtensionClient performance

### PR DESCRIPTION
**Pull request by ndinu**

### CODE REVIEW

**Comment by ndinu** [comment]
--validate

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/client/strategy/OperationMessageProcessorStrategy.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import org.mule.runtime.extension.api.client.OperationParameters;
+import org.mule.runtime.module.extension.internal.runtime.operation.OperationMessageProcessor;
+
+/**
+ * Interface that provides help for executing extension operations.
+ *
+ * @since 4.1.6
+ */
+@NoImplement
+public interface OperationMessageProcessorStrategy {
```
Comment (alegmarra) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/client/strategy/OperationMessageProcessorStrategy.java]: I'd rename OperationMessageProcessorStrategy to something much more specific to the EC, like ExtensionsClientProcessorsStrategy

Comment (ndinu) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/client/strategy/OperationMessageProcessorStrategy.java]: Agree

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/client/strategy/AbstractOperationMessageProcessorStrategy.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+/**
+ * Abstract base class that hold useful logic to implement a {@link OperationMessageProcessorStrategy}
+ *
+ * @since 4.1.6
+ */
+public abstract class AbstractOperationMessageProcessorStrategy implements OperationMessageProcessorStrategy {
+
+  protected final TemplateParser parser = TemplateParser.createMuleStyleParser();
+
+  protected ExtensionManager extensionManager;
```
[the remark this answers is not in the document]
Comment (ndinu) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/client/strategy/AbstractOperationMessageProcessorStrategy.java]: will change

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/client/ExtensionsClientTestCase.java [hunk 3]
```diff
@@ -234,4 +274,14 @@ public void disposeOnFailureOperation() throws Throwable {
     executeFailureOperation();
     assertThat(HeisenbergOperations.disposed, is(true));
   }
+
+  @Test
+  @Description("Using dynamic configurations, config parameters are resolved each time.")
+  public void executeTwiceWithDynamicConfiguration() throws Throwable {
+
```
[the remark this answers is not in the document]
Comment (ndinu) [reply modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/client/ExtensionsClientTestCase.java]: This was already covered in a functional test case, I already removed this empty test

**Pull request merged**